7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m

```diff
@@ -481,18 +481,29 @@ - (void)setBluetoothAsPrefferedInputDevice
 
 - (void) onInterruptionEvent:(NSNotification *) notification
 {
-    dispatch_async(_safetyQueue, ^() {
-        [self handleInterruptionEvent:notification];
-    });
-}
-
-- (void) handleInterruptionEvent:(NSNotification *) notification
-{
     NSDictionary *interruptionDict = notification.userInfo;
     NSInteger interruptionType =
     [[interruptionDict valueForKey:AVAudioSessionInterruptionTypeKey]
      integerValue];
-    
+
+    if (interruptionType == AVAudioSessionInterruptionTypeBegan)
+    {
+        dispatch_async(_safetyQueue, ^() {
+            [self handleInterruptionEvent:interruptionType];
+        });
+    }
+    else
+    {
+        /* 2 seconds delay gives time to release ios system apps
+         (e.g. phone call, siri, facetime) audio session */
```

Comment:
Are you positive there is no alternate way? This is sorta equivalent to putting sleep(2) into your code.